react-components/src/hooks/use3dScenes.tsx

```diff
@@ -12,97 +12,105 @@ import { type Cdf3dRevisionProperties } from './types';
-              properties.translationX,
-              properties.translationY,
-              properties.translationZ
-            );
-            transform.premultiply(translation);
-
-            transform.premultiply(CDF_TO_VIEWER_TRANSFORMATION);
-
-            const newModel = {
-              modelId: newModelId,
-              revisionId: newModelRevisionId,
-              transform
-            };
-
-            if (sceneModels !== undefined) {
-              sceneModels.push(newModel);
-            } else {
-              acc[externalId] = [newModel];
-            }
+          const newModelId = Number(edge.endNode.externalId);
+          const newModelRevisionId = Number(properties?.revisionId);
 
-            return acc;
-          },
-          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-          {} as Record<string, AddReveal3DModelOptions[]>
-        );
+          if (isNaN(newModelId) || isNaN(newModelRevisionId)) {
+            return;
+          }
 
-      return scenesMap;
+          const transform = new Matrix4();
+
+          transform.makeRotationFromEuler(
+            new Euler(
+              MathUtils.degToRad(properties.eulerRotationX),
+              MathUtils.degToRad(properties.eulerRotationY),
+              MathUtils.degToRad(properties.eulerRotationZ)
+            )
+          );
+
+          fixModelScale(properties);
+
+          const scaleMatrix = new Matrix4().makeScale(
+            properties.scaleX,
+            properties.scaleY,
+            properties.scaleZ
+          );
+          transform.multiply(scaleMatrix);
+
+          const translation = new Matrix4().makeTranslation(
+            properties.translationX,
+            properties.translationY,
+            properties.translationZ
+          );
+          transform.premultiply(translation);
+
+          transform.premultiply(CDF_TO_VIEWER_TRANSFORMATION);
+
+          const newModel = {
+            modelId: newModelId,
+            revisionId: newModelRevisionId,
+            transform
+          };
+
+          if (sceneModel !== undefined) {
+            sceneModel.addModelOptions.push(newModel);
+          } else {
+            sceneList.push({
+              externalId,
+              space,
+              addModelOptions: [newModel]
+            });
+          }
+        },
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        {} as Scene3DModel[]
+      );
+
+      return sceneList.reduce<Record<string, Scene3DModel>>((acc, scene) => {
+        acc[scene.externalId + '-' + scene.space] = scene;
+        return acc;
+      }, {});
     } catch (error) {
       console.warn("Scene space doesn't exist or has no scenes with 3D models");
       return {};
     }
   };
 
-  return useQuery<Record<string, AddReveal3DModelOptions[]>>(
-    ['cdf', '3d', 'scenes'],
-    queryFunction
-  );
+  return useQuery<Record<string, Scene3DModel>>(['cdf', '3d', 'scenes'], queryFunction);
```

Comment:
This should be prefixed with `reveal-react-components`. I believe we will formalize this a bit more in the future, but that would be the best for now